modal/_utils/async_utils.py

```diff
@@ -384,3 +392,40 @@ async def asyncnullcontext(*args, **kwargs):
         pass
     """
     yield
+
+
+def run_coro_sync(coro: typing.Awaitable[T]) -> T:
+    # more or less copied from synchronicity's implementation, but using current thread's event loop
+    loop = asyncio.get_event_loop()
+    if loop.is_running():
```

Comment:
Ah, it could definitely apply here. Nice heads up, I wasn't aware they had deprecated the implicit event loop creation...

Reply:
Ugh, not actually sure how to get similar behavior on 3.12 without the deprecation warning, unless we create a new event loop for every map call... which I guess in an option 😬 
`get_event_loop()` seems like the only (public) way to get the bound event loop of the current thread, but in this case we can't be sure if there is one or not, and I can't find a way to check that. `get_running_loop()` is only supported from calls within the event loop...

But it might actually not be a stupid idea to create a new event loop for the map call - it would actually get rid of the invalid error below as well :)

Reply:
Hmm so apparently you can't use event loops within other event loops, even explicitly without using hacky things like the nest-asyncio package...
Anyway, I refactored this to use `get_running_loop()` + `new_event_loop()` instead, which *should* not cause any 3.12 warnings. Also raising a more actionable error message to users if they accidentally use `.map()` from within an existing event loop